src/main/java/io/cryostat/recordings/RecordingTargetHelper.java

```diff
@@ -154,13 +157,16 @@ public IRecordingDescriptor startRecording(
                     Optional<IRecordingDescriptor> previous =
                             getDescriptorByName(connection, recordingName);
                     if (previous.isPresent()) {
-                        if (!restart) {
+                        if (!restartRecording) {
                             throw new IllegalArgumentException(
                                     String.format(
                                             "Recording with name \"%s\" already exists",
                                             recordingName));
-                        } else {
+                        } else if (isRecordingStopped(previous.get())) {
                             connection.getService().close(previous.get());
+                        } else {
+                            // if recording exists and still running, do nothing.
+                            return previous.get();
```

Comment:
I think this "do nothing" path should only happen if `replace == stopped`.

If `restart == true` or `replace == always` we should still restart the recording.

Reply:
I think it should restart when the recording is present and replace is stopped

Reply:
Could you update the unit tests or integration tests for these various cases?

Reply:
Hi Andrew, the shouldRestartRecording handles this when the recording is stopped and exists it should restart.. In other words, can you please guide me through one what tests should be edited to be updated to meet the restart and replace? I have tried manually to change the params of replace and restart and i see all is well

Reply:
Since the main change is in `RecordingTargetHelper`, which is referenced by both the V1 and GraphQL API endpoints, I think adding the bulk of the tests in `RecordingTargetHelperTest` would be suitable. I see that that class has been updated to simply replace `false` with `"never"`, but the other cases of `"stopped"` and `"always"` also need to be tested. Also, the behaviour of what happens when both `?replace` and `?restart` are provided can be tested.

Reply:
Sounds good.

Reply:
I'll take a look, but I think it makes sense that the test cases seem repetitive. Each test case will mostly be the same as the others with the same kinds of mocks put together and mock behaviours defined, but just with minor differences in the mock recording state, the replacement parameter value, and the assertion/verification of the result. It'll probably look like each test is 40 lines long and only 4 lines are different between them.